lib/ex_unit/test/ex_unit_test.exs

```diff
@@ -1101,6 +1101,48 @@ defmodule ExUnitTest do
     assert third =~ "ThirdTestFIFO"
   end
 
+  for async? <- [true, false] do
```

Comment:
Could we instead run the same modules twice, but with different arguments, in the same test? That should make it faster and simpler!

Reply:
yes we can, but I needed to adjust `run_with_filter`, see the new commit. I'm not sure if I like this more.